staging/src/k8s.io/cloud-provider/app/controllermanager.go

```diff
@@ -307,11 +307,11 @@ type InitCloudFunc func(config *cloudcontrollerconfig.CompletedConfig) cloudprov
 type InitFunc func(ctx genericcontrollermanager.ControllerContext) (debuggingHandler http.Handler, enabled bool, err error)
 
 // InitFuncConstructor is used to construct InitFunc
-type InitFuncConstructor func(completedConfig *cloudcontrollerconfig.CompletedConfig, cloud cloudprovider.Interface) InitFunc
+type InitFuncConstructor func(clientName string, completedConfig *cloudcontrollerconfig.CompletedConfig, cloud cloudprovider.Interface) InitFunc
```

Comment:
Ditto passing the ControllerInitializerConstructor

Reply:
Same comment as above, I refactored slightly and I think it achieves the same goal.